File: benchmark/fib.go
package main

import (
	"fmt"
	"time"
)

func fibonacci() func() int {
	a, b := 0, 1
	return func() int {
		result := a
		a, b = b, a+b
		return result
	}
}

func main() {
	start := time.Now()

	fib := fibonacci()
	for i := 0; i < 100000000; i++ {
		fib()
	}

	elapsed := time.Since(start)
	fmt.Printf("Execution time for Go (Golang): %s\n", elapsed)
}

File: benchmark/fib.js
function fibonacci() {
    let a = 0, b = 1;
    return function() {
        const result = a;
        const c = a + b;
        a = b;
        b = c;
        return result;
    };
}

console.time("nodejs");

const fib = fibonacci();
for (let i = 0; i < 100000000; i++) {
    fib();
}

console.timeEnd("nodejs");
